File: Solutions/Medium/3577. Count the Number of Computer Unlocking Permutations/Solution.cs
public class Solution {
    const int MOD = (int)1e9 + 7;
    public int CountPermutations(int[] complexity) {
        int firstComplexity = complexity[0], n = complexity.Length;
        for(int i = 1; i < n; i++){
            if(complexity[i] <= firstComplexity){
                return 0;
            }
        }
        return Factorial(n - 1, 1);
    }

    private int Factorial(int n, long result){
        if(n == 0) return (int)result;
        return Factorial(n - 1, (result * n) % MOD);
    }
}

File: Solutions/Medium/3577. Count the Number of Computer Unlocking Permutations/Solution.py
class Solution:
    MOD: int = 1000000007
    def countPermutations(self, complexity: List[int]) -> int:
        first_complexity: int = complexity[0]
        n: int = len(complexity)
        for i in range(1, n):
            if (first_complexity >= complexity[i]):
                return 0
        return self.modFact(n - 1, 1)

    def modFact(self, n: int, result: int) -> int:
        if (n == 0):
            return result
        return self.modFact(n - 1, (result * n) % self.MOD)